### PR TITLE
rework of PATBench after functional review with Brent

### PR DESCRIPTION
**Pull request by OliverHerms**
Notion list of changes:
https://www.notion.so/cowboyhq/PAT-Testbench-firmware-review-notes-3bf7645b1c044ab5bd28e971f8a7c83c

mainly error handling of use cases with broken boards.

### CODE REVIEW

File: PAT_bench.py [hunk 1]
```diff
@@ -16,43 +16,94 @@ def __init__(self, logger):
         #LOAD
         self.DL3021 = dl3021_logger()
```
Comment (LaurentErnes) [inline PAT_bench.py:17]: I'm not sure why it is named 'logger'. It seems to be the actual device, not just a logger ?

File: PAT_bench.py [hunk 2]
```diff
@@ -16,43 +16,94 @@ def __init__(self, logger):
...
+    ## RIGOL POWER SUPPLY
+    def dp832_set_channels(self):
+        self.DP832.set_channel( 1, 32, 0.8, 1) # #Channel, Voltage, Current, Current-Limit
+        self.DP832.set_channel( 2, 4, 0.25, 0.3) ##Channel, Voltage, Current, Current-Limit
+        if CONSOLE:    
+            print('-----')
+            print('power supply on')
+            print('-----')
+
+    def dp832_set_HighV(self):
```
Comment (LaurentErnes) [inline PAT_bench.py:54]: again `set_HighV()` would suffice

File: PAT_bench.py [hunk 3]
```diff
@@ -16,43 +16,94 @@ def __init__(self, logger):
         #LOAD
         self.DL3021 = dl3021_logger()
         self.DL3021.set_DEBUG(False)
-        self.ODP3033 = odp3033_logger()
-        self.ODP3033.set_DEBUG(False)
+        #self.ODP3033 = odp3033_logger()
+        #self.ODP3033.set_DEBUG(False)
+        self.DP832 = dp832_logger()
+        self.DP832.set_DEBUG(False)
```
Comment (LaurentErnes) [inline PAT_bench.py:22]: As they share the same API. To select one power supply or the other, you can just have:
```
if some_condition:
    self.power_supply = odp3033_logger
else:
    self.power_supply = dp832_logger
```
And then use the same `self.power_supply` object.
Also, you'll need to have a different set of parameters `self.power_supply_config = [...]` to take into account the possible channel / voltage /... difference.

Comment (LaurentErnes) [reply PAT_bench.py:22]: To avoid duplicating functions for every different PS.

File: PAT_bench.py [hunk 4]
```diff
@@ -16,43 +16,94 @@ def __init__(self, logger):
...
+    def opd3033_get_current(self, channel):
+        measure = self.ODP3033.measure_current(channel)
+        return measure
 
-    def dl3021_get_power(self):
-        power = self.DL3021.get_power()
-        return power
 
     def board_on(self):
         self.ODP3033.output_on(1)
```
Comment (LaurentErnes) [inline PAT_bench.py:109]: That won't work if `self.ODP3033` is not defined.

File: config.py [hunk 5]
```diff
@@ -0,0 +1,37 @@
+class values():
```
Comment (LaurentErnes) [inline config.py:1]: OK to group everything in one place. Even better would be to load this config from a JSON (or YAML) file: not hardcoded here.

File: main.py [hunk 6]
```diff
@@ -118,52 +81,67 @@ def main(args):
...
+            testcase.check_temperature_sensors()
+            testcase.check_registers()
+            testbench.dp832_set_HighV()
+            #testcase.check_board_current_high()
+            testcase.start_charging()
+            testcase.check_power()
+            testcase.stop_charging()
+            testcase.set_test_passed()
+            testcase.stop_timer()
+            testbench.all_instruments_off()
```
Comment (LaurentErnes) [inline main.py:119]: * Could be converted into a loop running tests from a list.
* `testbench` actions should be integrated into testcases, and not be seen here.

**Review by LaurentErnes — COMMENTED** [review]
I made a very rapid review, see comments.
Also, try not to have `print` and use systematically `logger.info`, `logger.warning`or `logger.error`: so nothing is lost and all is categorized.

**Review by paul-cowboy — COMMENTED** [review]
I second Laurent's comments

File: PAT_bench.py (hunk 3 above)
Comment (OliverHerms) [reply PAT_bench.py:22]: The Owon pwoer supply was a test, we switched back to the Rigol one, so this option and the duplication can be eliminated.

File: config.py (hunk 5 above)
Comment (OliverHerms) [reply config.py:1]: Ok, haven´t done that before. We can do that when reworking for the com module maybe.

File: PAT_bench.py [hunk 7]
```diff
@@ -3,11 +3,11 @@
 sys.path.insert(0, './hw_tools/testscripts/cockpit_c4_validation')
 
 #rearrange instruments into drivers
-#from hw_tools.testscripts.cockpit_c4_validation.rigol_DP832 import dp832_logger #define import location after folder structure in Repository is in place
+from hw_tools.testscripts.cockpit_c4_validation.rigol_DP832 import dp832_logger #define import location after folder structure in Repository is in place
 from hw_tools.testscripts.cockpit_c4_validation.rigol_DL3021 import dl3021_logger
 from hw_tools.testscripts.cockpit_c4_validation.owon_ODP3033 import odp3033_logger
 
-CONSOLE = False #monitor error messages
+CONSOLE = True #monitor error messages
```
[the remark this answers is not in the document]
Comment (OliverHerms) [reply PAT_bench.py:10]: The reason I decided not to use the logging.info function, is that it gets really messy in the console and I got a lot more messages than I wanted to. 

I understand the purpose of the different logging levels and for a logfile and a developer to debug what went wrong its great, but for someone working at Page on the PAT bench maybe not. 

I can add more logging.info to have all the information saved properly as well, but I designed the console as an Interface and wanted to visualize it myself, make it as appealing as it is possible in this setting. Did´t have time to also develop a GUI.

File: PAT_bench.py (hunk 4 above)
Comment (OliverHerms) [reply PAT_bench.py:109]: Forgot to change to `self.DL832`

File: PAT_bench.py (hunk 2 above)
Comment (OliverHerms) [reply PAT_bench.py:54]: renamed functions for instruments in PAT_bench.py

File: PAT_bench.py [hunk 8]
```diff
@@ -16,43 +16,94 @@ def __init__(self, logger):
...
+        #self.ODP3033.set_DEBUG(False)
+        self.DP832 = dp832_logger()
+        self.DP832.set_DEBUG(False)
         self.logger = logger
 
     ################################
     ## SETUP INSTRUMENT FUNCTIONS ##
     ################################
 
+    ## RIGOL LOAD
```
[the remark this answers is not in the document]
Comment (OliverHerms) [reply PAT_bench.py:29]: Added short documentation for PAT_bench.py and mentioned the used instruments there.

File: main.py (hunk 6 above)
Comment (OliverHerms) [reply main.py:119]: good point, moved `testbench` functions into the corresponding `testcase` functions.

We should discuss the while loop running the tests when reworking this for the more generic code.

File: PAT_bench.py [hunk 9]
```diff
@@ -16,43 +16,94 @@ def __init__(self, logger):
...
     def dl3021_turn_on(self, power):
```
[the remark this answers is not in the document]
Comment (OliverHerms) [reply PAT_bench.py:30]: renamed functions for the load and power supply in `PAT_bench.py`

File: PAT_bench.py (hunk 1 above)
Comment (OliverHerms) [reply PAT_bench.py:17]: true, was a leftover from previous code, where the logger was indeed part of the driver

**Comment by OliverHerms** [comment]
Is this review finished? 

Would like to request an updated version of the project.

**Review by LaurentErnes — APPROVED** [review]
I did not explicitly ask for changes since this is not the final source code yet. Waiting for the next PR.

**Pull request closed without merging**